From a3690aad663af51e846eaaa441d9b8f5dc3e8ad0 Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Tue, 20 Jan 2015 16:14:05 -0800
Subject: [PATCH 09/15] Enable wilink8 bluetooth chip for Overo

Add platform data for Gumstix Overo COMs to support bluetooth from the
Wilink WL18xx chips.  Here, board configuration data is encapsulated
in pdata-quirks.c rather than via device tree file as the DT-bindings
for this bluetooth modules are not yes in mainline kernels. We need to
disable the GPIO nReset for other bluetooth modules as the btwilink
wants to claim it.

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 arch/arm/boot/dts/omap3-overo-base.dtsi |  1 -
 arch/arm/mach-omap2/pdata-quirks.c      | 20 ++++++++++++++++++++
 2 files changed, 20 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/omap3-overo-base.dtsi b/arch/arm/boot/dts/omap3-overo-base.dtsi
index 717aaa7..714fba4 100644
--- a/arch/arm/boot/dts/omap3-overo-base.dtsi
+++ b/arch/arm/boot/dts/omap3-overo-base.dtsi
@@ -77,7 +77,6 @@
 		regulator-name = "regulator-w3cbw003c-bt-nreset";
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
-		gpio = <&gpio6 4 GPIO_ACTIVE_HIGH>;		/* gpio_164: BT nReset */
 		startup-delay-us = <10000>;
 	};
 };
diff --git a/arch/arm/mach-omap2/pdata-quirks.c b/arch/arm/mach-omap2/pdata-quirks.c
index 01a4746..67698c9 100644
--- a/arch/arm/mach-omap2/pdata-quirks.c
+++ b/arch/arm/mach-omap2/pdata-quirks.c
@@ -16,6 +16,7 @@
 #include <linux/of_platform.h>
 #include <linux/wl12xx.h>
 #include <linux/input/ft5x06_ts.h>
+#include <linux/ti_wilink_st.h>
 
 #include <linux/platform_data/pinctrl-single.h>
 #include <linux/platform_data/iommu-omap.h>
@@ -193,8 +194,27 @@ static void __init omap3_zoom_legacy_init(void)
 	legacy_init_wl12xx(WL12XX_REFCLOCK_26, 0, 162);
 }
 
+
+struct ti_st_plat_data wilink_pdata = {
+	.nshutdown_gpio = 164,
+	.dev_name = "/dev/ttyO1",
+	.flow_cntrl = 1,
+	.baud_rate = 115200,
+};
+static struct platform_device wl128x_device = {
+	.name		= "kim",
+	.id		= -1,
+	.dev.platform_data = &wilink_pdata,
+};
+static struct platform_device btwilink_device = {
+	.name = "btwilink",
+	.id = -1,
+};
+
 static void __init overo_legacy_init(void)
 {
+	platform_device_register(&btwilink_device);
+	platform_device_register(&wl128x_device);
 	legacy_init_wl12xx(WL12XX_REFCLOCK_26, 0, 58);
 }
 
-- 
1.9.1

